src/Auth/TokenResponse.js

```diff
@@ -0,0 +1,50 @@
+class TokenResponse{
```

Comment:
This object is not needed. You are converting a native structure (array) into an object, thus creating memory overhead. I would change this in the Authentication object as described in the comment from the OAuth object